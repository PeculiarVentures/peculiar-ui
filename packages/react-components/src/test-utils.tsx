import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
