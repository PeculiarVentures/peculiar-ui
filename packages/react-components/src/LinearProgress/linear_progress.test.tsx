import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { LinearProgress } from './index';

describe('<LinearProgress />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(<LinearProgress />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(<LinearProgress className="my-class-name" />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('colors', () => {
    const colors: React.ComponentProps<typeof LinearProgress>['color'][] = [
      'primary',
      'secondary',
    ];

    colors.forEach((color) => {
      it(`color "${color}"`, () => {
        const { asFragment } = render(
          <LinearProgress color={color} />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
