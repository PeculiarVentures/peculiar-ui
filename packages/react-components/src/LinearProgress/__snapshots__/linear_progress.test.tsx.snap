// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinearProgress /> colors color "primary" 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    left: -30%;
  }

  100% {
    left: 100%;
  }
}

.emotion-0 {
  background: var(--pv-color-primary-tint-4);
  height: 4px;
  overflow: hidden;
  position: relative;
}

.emotion-1 {
  background: var(--pv-color-primary);
  top: 0;
  left: 0;
  bottom: 0;
  width: 30%;
  position: absolute;
  transform-origin: left;
  -webkit-animation: animation-0 2s cubic-bezier(0.53, 0.21, 0.29, 0.67) 0s infinite;
  animation: animation-0 2s cubic-bezier(0.53, 0.21, 0.29, 0.67) 0s infinite;
}

<div
    class="emotion-0"
    role="progressbar"
  >
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;

exports[`<LinearProgress /> colors color "secondary" 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    left: -30%;
  }

  100% {
    left: 100%;
  }
}

.emotion-0 {
  background: var(--pv-color-secondary-tint-4);
  height: 4px;
  overflow: hidden;
  position: relative;
}

.emotion-1 {
  background: var(--pv-color-secondary);
  top: 0;
  left: 0;
  bottom: 0;
  width: 30%;
  position: absolute;
  transform-origin: left;
  -webkit-animation: animation-0 2s cubic-bezier(0.53, 0.21, 0.29, 0.67) 0s infinite;
  animation: animation-0 2s cubic-bezier(0.53, 0.21, 0.29, 0.67) 0s infinite;
}

<div
    class="emotion-0"
    role="progressbar"
  >
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;

exports[`<LinearProgress /> should pass className 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    left: -30%;
  }

  100% {
    left: 100%;
  }
}

.emotion-0 {
  background: var(--pv-color-primary-tint-4);
  height: 4px;
  overflow: hidden;
  position: relative;
}

.emotion-1 {
  background: var(--pv-color-primary);
  top: 0;
  left: 0;
  bottom: 0;
  width: 30%;
  position: absolute;
  transform-origin: left;
  -webkit-animation: animation-0 2s cubic-bezier(0.53, 0.21, 0.29, 0.67) 0s infinite;
  animation: animation-0 2s cubic-bezier(0.53, 0.21, 0.29, 0.67) 0s infinite;
}

<div
    class="my-class-name emotion-0"
    role="progressbar"
  >
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;

exports[`<LinearProgress /> should render with default styles 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    left: -30%;
  }

  100% {
    left: 100%;
  }
}

.emotion-0 {
  background: var(--pv-color-primary-tint-4);
  height: 4px;
  overflow: hidden;
  position: relative;
}

.emotion-1 {
  background: var(--pv-color-primary);
  top: 0;
  left: 0;
  bottom: 0;
  width: 30%;
  position: absolute;
  transform-origin: left;
  -webkit-animation: animation-0 2s cubic-bezier(0.53, 0.21, 0.29, 0.67) 0s infinite;
  animation: animation-0 2s cubic-bezier(0.53, 0.21, 0.29, 0.67) 0s infinite;
}

<div
    class="emotion-0"
    role="progressbar"
  >
    <div
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;
